frontend/packages/ceph-storage-plugin/integration-tests/tests/2-tests/add-capacity.scenario.ts

```diff
@@ -108,60 +89,175 @@ if (clusterStatus && cephHealth) {
 
-      await clickKebabAction(uid, 'Add Capacity');
-      await verifyFields();
+      // eslint-disable-next-line no-useless-escape
+      expansionObjects.defaultSC = execSync(
+        `kubectl get storageclasses | grep -Po '\\w+(?=.*default)'`,
+      )
+        .toString()
+        .trim();
+      await click(getSCOption(expansionObjects.defaultSC));
+      verifyFields();
       await click(confirmButton);
 
-      const statusCol = storageClusterRow(uid).$('td:nth-child(3)');
-
-      // need to wait as cluster states fluctuates for some time. Waiting for 2 secs for the same
-      await browser.sleep(2 * SECOND);
-
-      await browser.wait(until.textToBePresentInElement(statusCol, CLUSTER_STATUS.PROGRESSING));
-      await browser.wait(
-        until.textToBePresentInElement(
-          statusCol.$('span.co-icon-and-text span'),
-          CLUSTER_STATUS.READY,
-        ),
-      );
+      await browser.sleep(5 * SECOND);
 
       expansionObjects.updatedClusterJSON = JSON.parse(
-        execSync(`kubectl get -o json -n ${NS} ${KIND} ${name}`).toString(),
+        execSync(`kubectl get -o json -n ${NS} ${KIND} ${expansionObjects.name}`).toString(),
       );
       expansionObjects.updatedCnt = _.get(
         expansionObjects.updatedClusterJSON,
         'spec.storageDeviceSets[0].count',
       );
-      expansionObjects.updatedPods = JSON.parse(
-        execSync(`kubectl get pod -o json -n ${NS}`).toString(),
+    });
+
+    it('Newly added capacity should takes into effect at the storage level', () => {
+      // by default 2Tib capacity is being added
+      expect(expansionObjects.updatedCnt - expansionObjects.previousCnt).toEqual(1);
+    });
+
+    it('Selected storage class should be sent in the YAML', () => {
+      const storageCR = JSON.parse(
+        execSync(`kubectl get storageclusters ${STORAGE_CLUSTER_NAME} -n ${NS} -o json`).toString(),
       );
+      const scFromYAML =
+        storageCR?.spec?.storageDeviceSets?.[0]?.dataPVCTemplate?.spec?.storageClassName;
+      expect(expansionObjects.defaultSC).toEqual(scFromYAML);
+    });
+  });
 
-      // need to wait to get the new osds reflected in osd tree
+  describe('Addition tests for Baremetal infra', () => {
+    beforeAll(async () => {
+      await selectSCDropdown(expansionObjects.uid);
+      execSync(`echo '${JSON.stringify(storageClass)}' | oc apply -f -`);
+      // need to wait for some time in order to reflect the storage class
       await browser.sleep(1 * MINUTE);
+      await click(getSCOption(storageClass.metadata.name));
+    });
 
-      expansionObjects.updatedOSDTree = JSON.parse(
-        execSync(
-          `oc -n ${NS} rsh  $(oc -n ${NS} get pod | grep ceph-operator| awk '{print$1}') ceph --conf=/var/lib/rook/${NS}/${NS}.config osd tree --format=json`,
-        ).toString(),
-      );
+    afterAll(async () => {
+      execSync(`echo '${JSON.stringify(storageClass)}' | oc delete -f -`);
+    });
```

Comment:
Don't mix Baremetal tests and OCS tests.

Reply:
Baremetal tests are subset of ocs tests as apart from few specific checks for baremetal, rest are common for all infra.

Reply:
But these tests would fail in a cloud deployment right?

Reply:
No, it will not as I have mocked the flow by creating SC for bare metal to test this and deleting post checks.